src/components/SearchInput/SearchInput.jsx

```diff
@@ -1,7 +1,63 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import { DebounceInput } from 'react-debounce-input';
+
+import styles from './SearchInput.module.scss';
+import SearchIcon from '../../assets/images/search-icon.png';
+import SearchLocationIcon from '../../assets/images/search-location-icon.png';
+import { NUMBER_OF_RESULTS } from '../../constants/forecaApi';
+import { dataService } from '../../services/dataService';
 
 function SearchInput() {
-  return <div>Search input</div>;
+  const [searchQuery, setSearchQuery] = useState('');
+  const [searchResults, setSearchResults] = useState([]);
+
+  useEffect(() => {
+    if (searchQuery) {
+      searchCities(searchQuery);
+    }
+  }, [searchQuery]);
+
+  async function searchCities(query) {
+    try {
+      const { locations } = await dataService.searchCity(query);
+
+      setSearchResults(locations);
+    } catch (error) {
+      console.log(error);
+    }
+  }
+
+  const handleChange = e => {
+    setSearchQuery(e.target.value);
+  };
+
+  const renderSearchResults = searchResults.map(item => (
```

Comment:
I don't think there is a lot of added value to move this mapping here. It is super usual to write it "inline" right n JSX

Reply:
ok, got it